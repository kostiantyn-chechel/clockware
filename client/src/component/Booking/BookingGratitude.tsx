import React from 'react';
import Typography from '@material-ui/core/Typography';
import { hoursToWords } from '../../helpers/dateTime';
import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles((theme) => ({
    text: {
        marginTop: theme.spacing(3),
    },
}));

export interface IBookingGratitude {
    masterName: string,
    date: string,
    time: string,
    size: string,
    email: string,
}

const BookingGratitude:React.FC<IBookingGratitude> = (props) => {
    const classes = useStyles();

    return (
        <React.Fragment>
            <Typography component="h1" variant="h4" align="center" color="textPrimary">
                Спасибо за Ваш заказ!
            </Typography>
            <Typography id={'gr-master'} className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мастер {props.masterName} ждет Вас
            </Typography>
            <Typography id={'gr-time'} className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                {props.date} в {props.time}
            </Typography>
            <Typography id={'gr-hours'} className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Ремонт займет {hoursToWords(props.size)}
            </Typography>
            <Typography id={'gr-email'} className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мы выслали напоминание о визите на Ваш email:  {props.email}
            </Typography>
        </React.Fragment>
    );
}

export default BookingGratitude;