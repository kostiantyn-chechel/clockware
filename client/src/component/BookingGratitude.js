import React from 'react';
import Typography from '@material-ui/core/Typography';
import {hoursByWords} from '../helpers/dateTime';
import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles((theme) => ({
    text: {
        marginTop: theme.spacing(3),
    },
}));

function BookingGratitude(props) {
    const classes = useStyles();

    return (
        <React.Fragment>
            <Typography component="h1" variant="h4" align="center" color="textPrimary">
                Спасибо за Ваш заказ!
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мастер {props.masterName} ждет Вас
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                {props.date} в {props.time}
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Ремонт займет {hoursByWords(props.size)}
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мы выслали напоминание о визите на Ваш email:  {props.email}
            </Typography>
        </React.Fragment>
    );
}

export default BookingGratitude;