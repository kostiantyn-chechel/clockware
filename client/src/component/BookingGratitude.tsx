import React from 'react';
import Typography from '@material-ui/core/Typography';
import { hoursToWords } from '../helpers/dateTime';
import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles((theme) => ({
    text: {
        marginTop: theme.spacing(3),
    },
}));

interface IBookingGratitude {
    masterName: string,
    date: string,
    time: string,
    size: string,
    email: string,
}

const BookingGratitude:React.FC<IBookingGratitude> = (props) => {
    const classes = useStyles();

    return (
        <React.Fragment>
            <Typography component="h1" variant="h4" align="center" color="textPrimary">
                Спасибо за Ваш заказ!
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мастер {props.masterName} ждет Вас
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                {props.date} в {props.time}
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Ремонт займет {hoursToWords(props.size)}
            </Typography>
            <Typography className={classes.text} component="h2" variant="h5" align="center" color="textPrimary">
                Мы выслали напоминание о визите на Ваш email:  {props.email}
            </Typography>
        </React.Fragment>
    );
}

export default BookingGratitude;