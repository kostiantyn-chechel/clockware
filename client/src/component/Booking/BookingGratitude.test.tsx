import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import BookingGratitude, { IBookingGratitude } from './BookingGratitude';

describe('Test <BookingGratitude>', () => {
    const props: IBookingGratitude = {
        masterName: 'Vasia Vasia',
        date: '2021-02-02',
        time: '15:00',
        size: '2',
        email: 'vasia@vasia.com'
    };

    const Component = <BookingGratitude {...props}/>;
    const ShallowComponent = shallow(Component);

    test('BookingGratitude component should exists', () => {
        expect(ShallowComponent).toBeDefined();
    });

    test("ToMatchSnapshot BookingGratitude with default date", () => {
        expect(renderer.create(Component).toJSON()).toMatchSnapshot();
    });

});