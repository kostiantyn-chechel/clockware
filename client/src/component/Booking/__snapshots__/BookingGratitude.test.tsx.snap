// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <BookingGratitude> ToMatchSnapshot BookingGratitude with default date 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
  >
    Спасибо за Ваш заказ!
  </h1>,
  <h2
    className="MuiTypography-root makeStyles-text-1 MuiTypography-h5 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    id="gr-master"
  >
    Мастер 
    Vasia Vasia
     ждет Вас
  </h2>,
  <h2
    className="MuiTypography-root makeStyles-text-1 MuiTypography-h5 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    id="gr-time"
  >
    2021-02-02
     в 
    15:00
  </h2>,
  <h2
    className="MuiTypography-root makeStyles-text-1 MuiTypography-h5 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    id="gr-hours"
  >
    Ремонт займет 
    два часа
  </h2>,
  <h2
    className="MuiTypography-root makeStyles-text-1 MuiTypography-h5 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    id="gr-email"
  >
    Мы выслали напоминание о визите на Ваш email:  
    vasia@vasia.com
  </h2>,
]
`;
